bson/src/main/org/bson/codecs/configuration/CodecRegistry.java

```diff
@@ -65,6 +65,9 @@ public interface CodecRegistry extends CodecProvider {
      * @since 4.8
      */
     default <T> Codec<T> get(Class<T> clazz, List<Type> typeArguments) {
-        throw new CodecConfigurationException("Make this message really informative");  // TODO
+        throw new CodecConfigurationException("The application is configuring a codec for a parameterized class, but is using an "
+                + "implementation of org.bson.codecs.configuration.CodecRegistry that does not override the default implementation of the"
+                + "method get(java.lang.Class<T>, java.util.List<java.lang.reflect.Type>).  This implementation should be replaced by one "
+                + "that is obtained from one of the CodecRegistry builder methods in org.bson.codecs.configuration.CodecRegistries");
```

Comment:
Given the [discussion above](https://github.com/mongodb/mongo-java-driver/pull/1007#discussion_r992378186), we don't need this method to have a default implementation, right?

Reply:
I think it's still a good idea, just in case there are applications that are (unnecessarily) creating their own CodecRegistry implementations only to have them treated as CodecProvider.  This will avoid unnecessary compilation errors for that scenario.

Reply:
Are there situations when a user may see the error message from the default implementation? Unless I am confused, that can't happen. If that's the case, we can just use `Assertions.fail()` as the default implementation.

Reply:
Yeah, I agree. This message is wrong given what we know now.

Reply:
Done